marimo/_runtime/packages/package_manager.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2024 Marimo. All rights reserved.
+from __future__ import annotations
+
+import abc
+
+
+class PackageManager(abc.ABC):
+    """Interface for a package manager that can install packages."""
+
+    @abc.abstractmethod
+    def module_to_package(self, module_name: str) -> str:
+        """Canonicalizes a module name to a package name."""
+        ...
+
+    @abc.abstractmethod
+    def package_to_module(self, package_name: str) -> str:
+        """Canonicalizes a package name to a module name."""
+        ...
+
+    @abc.abstractmethod
+    async def install(self, package: str) -> bool:
```

Comment:
is it worth installing multiple at once? its more performant and possibly more correct because of version compatibility

Reply:
> possibly more correct because the package manager can find the right versions compatible across all the packages requested to be installed

Good point regarding correctness. Unfortunately we may lose the progress notifications in the UI, though, if we install multiple at once.